x-pack/plugins/task_manager/server/buffered_task_store.ts

```diff
@@ -60,6 +68,25 @@ export class BufferedTaskStore implements Updatable {
     });
   }
 
+  public async partialUpdate(
+    partialDoc: PartialConcreteTaskInstance,
+    options: { validate: boolean; doc: ConcreteTaskInstance }
+  ): Promise<ConcreteTaskInstance> {
+    // merge the partial updates with the doc and validate
+    this.taskStore.taskValidator.getValidatedTaskInstanceForUpdating(
+      { ...options.doc, ...partialDoc },
+      { validate: options.validate }
+    );
+
+    const result = await unwrapPromise(this.bufferedPartialUpdate(partialDoc));
+
+    // merge the partial update result with the doc and validate
+    return this.taskStore.taskValidator.getValidatedTaskInstanceFromReading(
```

Comment:
cc @mikecote Do you think validating like this is ok?

Reply:
@ymao1 yeah, that's as good as I imagined. We can improve on it later on if we think of something else 👍